vendor/engines/secure_rooms/app/models/secure_rooms/card_reader.rb

```diff
@@ -4,6 +4,9 @@ class CardReader < ActiveRecord::Base
 
     belongs_to :secure_room, foreign_key: :product_id
 
+    validates :product_id, :card_reader_number, :control_device_number, presence: true
+    validates :card_reader_number, uniqueness: [:control_device_number]
```

Comment:
I've never seen this style before, and I'm not seeing a reference for it. I would expect it to be `validates :card_reader_number, uniqueness: { scope: :control_device_number }`.